tests/src/test/java/org/wildfly/plugins/bootablejar/maven/goals/AbstractBootableJarMojoTestCase.java

```diff
@@ -0,0 +1,364 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.plugins.bootablejar.maven.goals;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.maven.plugin.Mojo;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.galleon.config.ConfigId;
+import org.jboss.galleon.config.ConfigModel;
+import org.jboss.galleon.config.ProvisioningConfig;
+import org.jboss.galleon.util.PathsUtils;
+import org.jboss.galleon.util.ZipUtils;
+import org.jboss.galleon.xml.ProvisioningXmlParser;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.wildfly.core.launcher.ProcessHelper;
+import org.wildfly.plugin.core.ServerHelper;
+
+/**
+ * @author jdenise
+ */
+// Still using JUnit3 in AbstractMojoTestCase class.
```

Comment:
Is this comment still valid? Just asking because I can't find where this class is used here and I'd like to be sure to not be missing anything.

Reply:
I believe so, but I'm not really sure. The annotation should be inherited by the sub-classes. I did not try running without it TBH. I tried to not really change much in the tests themselves other than refactoring them to move each test to it's own class.

Reply:
Yes, this annotation is needed to be able to use @Test annotation.